WooCommerce/Classes/Authentication/Jetpack Setup/Site Credential Login/SiteCredentialLoginViewModel.swift

```diff
@@ -66,32 +61,64 @@ private extension SiteCredentialLoginViewModel {
             .map { $0.isEmpty || $1.isEmpty }
             .assign(to: &$primaryButtonDisabled)
     }
-}
 
-// MARK: LoginFacadeDelegate conformance
-//
-extension SiteCredentialLoginViewModel: LoginFacadeDelegate {
-    func displayRemoteError(_ error: Error) {
-        isLoggingIn = false
+    func loginAndAttemptFetchingJetpackPluginDetails() {
+        guard let loginURL = URL(string: siteURL + Constants.loginPath),
+              let adminURL = URL(string: siteURL + Constants.adminPath) else {
+            DDLogWarn("⚠️ Cannot construct login URL and admin URL for site \(siteURL)")
+            return
+        }
+
+        // Makes sure the loading indicator is shown
+        isLoggingIn = true
+
+        // Prepares the authenticator with username and password
+        let authenticator = CookieNonceAuthenticator(username: username,
+                                                     password: password,
+                                                     loginURL: loginURL,
+                                                     adminURL: adminURL,
+                                                     version: Constants.defaultWPVersion,
+                                                     nonce: nil)
+        let network = WordPressOrgNetwork(authenticator: authenticator)
+        let authenticationAction = JetpackConnectionAction.authenticate(siteURL: siteURL, network: network)
+        stores.dispatch(authenticationAction)
```

Comment:
Super nit: Could we extract this into a method like `func performCookieAuthentication`?

Reply:
Good point - updated in 65102068219eb97be64a17e02e35690c9ec5a068.